src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -592,83 +664,65 @@ CreateOffer::applyGuts (ApplyView& view, ApplyView& view_cancel)
 
     auto const sleCreator = view.peek (keylet::account(account_));
 
-    deprecatedWrongOwnerCount_ = sleCreator->getFieldU32(sfOwnerCount);
+    deprecatedWrongOwnerCount_ = (*sleCreator)[sfOwnerCount];
 
-    std::uint32_t const uAccountSequenceNext = sleCreator->getFieldU32 (sfSequence);
-    std::uint32_t const uSequence = tx().getSequence ();
+    auto const uAccountSequenceNext = (*sleCreator)[sfSequence];
+    auto const uSequence = tx().getSequence ();
 
     // This is the original rate of the offer, and is the rate at which
     // it will be placed, even if crossing offers change the amounts that
     // end up on the books.
-    std::uint64_t const uRate = getRate (saTakerGets, saTakerPays);
-
-    auto viewJ = ctx_.app.journal ("View");
+    auto const uRate = getRate (saTakerGets, saTakerPays);
 
-    TER result = tesSUCCESS;
+    auto viewJ = ctx_.app.journal("View");
 
-    // This is the ledger view that we work against. Transactions are applied
-    // as we go on processing transactions.
+    auto result = tesSUCCESS;
 
-    if (isGlobalFrozen (view, uPaysIssuerID) || isGlobalFrozen (view, uGetsIssuerID))
-    {
-        if (j_.warning) j_.warning <<
-            "Offer involves frozen asset";
-
-        result = tecFROZEN;
-    }
-    else if (accountFunds(view, account_, saTakerGets,
-        fhZERO_IF_FROZEN, viewJ) <= zero)
-    {
-        if (j_.debug) j_.debug <<
-            "delay: Offers must be at least partially funded.";
-
-        result = tecUNFUNDED_OFFER;
-    }
-    // This can probably be simplified to make sure that you cancel sequences
-    // before the transaction sequence number.
-    else if (bHaveCancel && (uAccountSequenceNext - 1 <= uCancelSequence))
-    {
-        if (j_.debug) j_.debug <<
-            "uAccountSequenceNext=" << uAccountSequenceNext <<
-            " uOfferSequence=" << uCancelSequence;
-
-        result = temBAD_SEQUENCE;
-    }
-
-    if (result != tesSUCCESS)
-    {
-        j_.debug << "final result: " << transToken (result);
-        return { result, true };
-    }
+    // Should have been checked and failure
+    // TER returned in preclaim if any of these
+    // are false.
+    assert(!isGlobalFrozen(view, uPaysIssuerID)
+        && !isGlobalFrozen(view, uGetsIssuerID));
+    assert(accountFunds(view, account_, saTakerGets,
+        fhZERO_IF_FROZEN, viewJ) > zero);
+    assert(!cancelSequence ||
+        (uAccountSequenceNext - 1 > *cancelSequence));
```

Comment:
> The asserts just duplicate the same logic that is in `preclaim`

That's exactly why those asserts are there. I left them to reassure myself that I implemented and called `preclaim` correctly, and once that was done, it seemed that there was no harm in leaving them there, since they won't be included in release code. For that reason, I would be against changing them to `throw`s.